src/containers/UserActionsBox/UserActionsBox.tsx

```diff
@@ -1,27 +1,61 @@
 import React from 'react'
 import '../../styles/UserActionsBox.scss'
-import Delete from '../../components/Delete/Delete'
+import Delete from '../../components/Delete/Delete';
+import {flagQuestion, flagComment} from '../../utils/util'
+import FlagModal from '../../components/flagModal/FlagModal'
+import { UserContext } from '../../contexts/UserContext';
+import {useContext} from 'react';
 
 interface UserActonsBoxProps {
   delete: (id: number) => void;
   id: number;
   update: () => void;
   updateDeleteStatus: () => void;
+  type: string;
 }
 
 const UserActionsBox: React.FC<UserActonsBoxProps> = (props) => {
+  const { userData } = useContext(UserContext);
 
   const handleDeleteClick = (event: React.FormEvent) => {
     props.delete(props.id)
     props.update()
     props.updateDeleteStatus()
   }
 
+const packagePost = (type: string, comment: string) => {
+if(type === "question") {
+  return {
+    "post": props.id, 
+    "user": userData.id,
+    "comment": comment 
+  }
+} else if(type === "comment")
+  return {
+    "response" : props.id,
+    "user" : userData.id,
+    "comment" : comment
+  }
+}
```

Comment:
I really like the use of an if statement here for packaging the post. Very smart way to do it.